### ISSUE TITLE
Outer JOINs wrapped as a subquery error out

### ISSUE THREAD

**Issue by saicitus** [citusdata/citus#2739]
```
create table test_a(id int);
CREATE TABLE
citus=> create table test_b(id int);
CREATE TABLE
citus=> select create_distributed_table('test_a','id');
 create_distributed_table
--------------------------

(1 row)

citus=> select create_distributed_table('test_b','id');
 create_distributed_table
--------------------------

(1 row)

citus=> select * from test_a a left join test_b b on a.id=b.id;
 id | id
----+----
(0 rows)

citus=> select * from (select * from test_a a left join test_b b on a.id=b.id)lo;
ERROR:  cannot perform distributed planning on this query
DETAIL:  Cartesian products are currently unsupported
```
Such queries could be generated by an ORM

**Comment by saicitus** [comment]
2 customers ran into this issue on the same day (today)

**Comment by onderkalaci** [comment]
That looks like a regression from 8.1 to 8.2 (e.g., this used to work in 8.1). @metdos is there anyone avaliable to work on within 8.3 cycle?

This [check](https://github.com/citusdata/citus/blob/master/src/backend/distributed/planner/query_pushdown_planning.c#L136-L143) should have detected to use query pushdown. However, it fails because we're only looking into `originalQuery->jointree` where the original query contains only a subquery, which is pulled up in the `rewrittenQuery`.

Just FYI for the readers, the same query is expected to fail [here](https://github.com/citusdata/citus/blob/master/src/backend/distributed/planner/query_pushdown_planning.c#L105-L115) as well, which is fine I think.

I think we should consider backporting it to 8.2 as well.

**Comment by colton-citus** [comment]
Is there an ETA on either an 8.2 patch to fix this or the 8.3 release? This is holding up testing for a prospect I'm working with.

**Comment by serprex** [comment]
@colton-citus #2742 should be merged by end of week, currently going through review process, and reviewers are almost ok with the change. It will be backported to 8.2 too.
